core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala

```diff
@@ -234,19 +247,25 @@ class ContainerProxy(factory: (TransactionId,
                      collectLogs: LogsCollector,
                      instance: InvokerInstanceId,
                      poolConfig: ContainerPoolConfig,
+                     healtCheckConfig: ContainerProxyHealthCheckConfig,
                      unusedTimeout: FiniteDuration,
                      pauseGrace: FiniteDuration)
     extends FSM[ContainerState, ContainerData]
     with Stash {
   implicit val ec = context.system.dispatcher
   implicit val logging = new AkkaLogging(context.system.log)
+  implicit val ac = context.system
+  implicit val materializer = ActorMaterializer()
   var rescheduleJob = false // true iff actor receives a job but cannot process it because actor will destroy itself
   var runBuffer = immutable.Queue.empty[Run] //does not retain order, but does manage jobs that would have pushed past action concurrency limit
   //track buffer processing state to avoid extra transitions near end of buffer - this provides a pseudo-state between Running and Ready
   var bufferProcessing = false
 
   //keep a separate count to avoid confusion with ContainerState.activeActivationCount that is tracked/modified only in ContainerPool
   var activeCount = 0;
+  var healthPingActor: Option[ActorRef] = None //setup after prewarm starts
+  //track the resuming run for easily referring to the action being resumed (it may fail)
+  var resumeRun: Option[Run] = None
```

Comment:
How about including this in FSMData?
I think that may make this more testable as we can set FSM data directly in the test codes.

Reply:
Do you mean make it a var on WarmedData?

Reply:
I meant `Option` data in FSMData(Warmed or not).

Reply:
I'm still not sure what you mean, I guess I need more context. I think it only applies on Warmed data, so it could possibly go there. But it doesn't apply always on WarmedData, which is why I kept it out of the state data, and just as part of the actor state.

Reply:
What I meant is just to include it in FSMData so that we can set the state and corresponding data with `TestFSMRef.setState()`. It will make this more testable.

If it only applies to `WarmedData` we can add it to `WarmedData`.
If it applies to multiple data, then we can just add it to multiple data.
I feel it is nicer to keep it in FSMData instead of having a `var` state.
We can easily transfer it between each transition with `stay with` directive.

Reply:
Current tests do not use TestFSMRef, rather they test the expected messages in and out of the actor. I moved that state to WarmedData and added tests - let me know if you have concerns.